aiortc/rtcdatachannel.py

```diff
@@ -17,17 +17,28 @@ class RTCDataChannel(EventEmitter):
     :param: parameters: An :class:`RTCDataChannelParameters`.
     """
 
-    def __init__(self, transport, parameters, id=None):
+    def __init__(self, transport, parameters, send_open=True):
         super().__init__()
         self.__bufferedAmount = 0
         self.__bufferedAmountLowThreshold = 0
-        self.__id = id
+        self.__id = parameters.id
         self.__parameters = parameters
         self.__readyState = 'connecting'
         self.__transport = transport
+        self.__send_open = send_open
+
+        if self.__parameters.negotiated and (
```

Comment:
I hadn't spotted this, but yes, absolutely yes! Thanks for being so thorough.

Reply:
As I am thinking now _self.__send_open_ is useless too. No need to store it.